### PR TITLE
Updates to run more MRI cext tests

### PR DESCRIPTION
**Pull request by bjfish**
This update moves the generated su files to `./ext/c` similar to MRI's `./ext/-test-`.

### CODE REVIEW

File: lib/cext/ruby.h [hunk 1]
```diff
@@ -70,6 +70,10 @@ extern "C" {
 
 #define HAVE_RB_IO_T 1
 
+#ifdef RUBY_EXTCONF_H
+#include RUBY_EXTCONF_H
+#endif
```
Comment (bjfish) [inline lib/cext/ruby.h]: Used to compile a number of tests that add a function via `extconf.h`:
https://github.com/graalvm/truffleruby/blob/ec915bdcf1cccba94d60fd31c5f4b12d52c83919/test/mri/tests/cext/c/hash/extconf.rb#L7

File: tool/jt.rb [hunk 2]
```diff
@@ -653,8 +654,12 @@ def compile_cext(name, ext_dir, target, *clang_opts)
     Dir.chdir(ext_dir) do
       STDERR.puts "in #{ext_dir} ..."
       run('-rmkmf', "extconf.rb") # -rmkmf is required for C ext tests
-      raw_sh("make")
-      FileUtils::Verbose.cp("#{name}.su", target) if target
+      if File.exists?('Makefile')
```
Comment (bjfish) [inline tool/jt.rb]: Added this to handle an extesion test that doesn't run on OSX: 
https://github.com/graalvm/truffleruby/blob/ec915bdcf1cccba94d60fd31c5f4b12d52c83919/test/mri/tests/cext/c/popen_deadlock/extconf.rb#L3

File: tool/jt.rb [hunk 3]
```diff
@@ -783,7 +788,7 @@ def run_mri_tests(extra_args, test_files, run_options = {})
     cext_tests = test_files.select { |f| f.include?("cext/ruby") }
     cext_tests.each do |test|
       test_path = "#{TRUFFLERUBY_DIR}/test/mri/tests/#{test}"
-      match = File.read(test_path).match(/^require ['"]c\/(.*?)["']/)
+      match = File.read(test_path).match(/\brequire ['"]c\/(.*?)["']/)
```
Comment (bjfish) [inline tool/jt.rb]: Not all requires start at the beginning of the line in the tests.

File: tool/jt.rb [hunk 4]
```diff
@@ -794,14 +799,22 @@ def run_mri_tests(extra_args, test_files, run_options = {})
                       else
                         "#{MRI_TEST_CEXT_DIR}/#{match[1]}"
                       end
-        compile_cext("#{match[1].split("/")[1]}", compile_dir, nil)
+        name = match[1].split('/').last
```
Comment (eregon) [inline tool/jt.rb]: Looks like File.basename

File: .gitignore [hunk 5]
```diff
@@ -27,6 +27,7 @@ lib/cext/trufflemock.bc
 lib/mri/openssl.su
 lib/mri/zlib.su
 lib/mri/syslog.su
+.ext
```
Comment (eregon) [inline .gitignore]: /.ext to only match at the repo root

File: lib/cext/ruby.h (hunk 1 above)
Comment (eregon) [inline lib/cext/ruby.h]: I assume MRI's ruby.h contains the same?

File: tool/jt.rb [hunk 6]
```diff
@@ -24,6 +24,7 @@
 TRUFFLERUBY_DIR = File.expand_path('../..', File.realpath(__FILE__))
 M2_REPO         = File.expand_path('~/.m2/repository')
 MRI_TEST_CEXT_DIR = "#{TRUFFLERUBY_DIR}/test/mri/tests/cext/c"
+MRI_TEST_CEXT_LIB_DIR = "#{TRUFFLERUBY_DIR}/.ext/c"
```
Comment (eregon) [inline tool/jt.rb]: Is this .ext where MRI expects to find the compiled c exts?

Comment (bjfish) [inline tool/jt.rb]: MRI has a `/.ext/-test-/` dir to match their requires. I'm not sure when or why our requires are different.
https://github.com/ruby/ruby/blob/1fe73fc948c5f511d3c6a7b4dfb7fe61348ab169/test/-ext-/hash/test_delete.rb#L3

File: lib/cext/ruby.h (hunk 1 above)
Comment (bjfish) [inline lib/cext/ruby.h]: @eregon Yes.

File: tool/jt.rb (hunk 4 above)
Comment (bjfish) [inline tool/jt.rb]: Yes, fixed.

File: .gitignore (hunk 5 above)
Comment (bjfish) [inline .gitignore]: Fixed.

File: tool/jt.rb (hunk 6 above)
Comment (eregon) [inline tool/jt.rb]: It's probably due to tool/update-mri.sh

**Pull request merged**